# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/shared/LIR-shared.h
# Commit: ecb8f8821de2
# Full Hash: ecb8f8821de2ab318eb4a0d863b5d9c1cd97d70f
# Author: Tooru Fujisawa <arai_a@mac.com>
# Date: 2016-04-07 20:42:20
# Regressor Bug: 1165052
# File Overlap Count: 1
# Description:
#   Bug 1165052 - Part 4: Inline IsConstructor intrinsic. r=jandem
# ==============================================================================

diff -r ae46f7e09a22 -r ecb8f8821de2 js/src/jit/shared/LIR-shared.h
--- a/js/src/jit/shared/LIR-shared.h	Wed Dec 23 12:46:35 2015 +0900
+++ b/js/src/jit/shared/LIR-shared.h	Mon Nov 16 09:01:37 2015 +0900
@@ -7215,6 +7215,22 @@
     }
 };
 
+class LIsConstructor : public LInstructionHelper<1, 1, 0>
+{
+  public:
+    LIR_HEADER(IsConstructor);
+    explicit LIsConstructor(const LAllocation& object) {
+        setOperand(0, object);
+    }
+
+    const LAllocation* object() {
+        return getOperand(0);
+    }
+    MIsConstructor* mir() const {
+        return mir_->toIsConstructor();
+    }
+};
+
 class LIsObject : public LInstructionHelper<1, BOX_PIECES, 0>
 {
   public: